lib/rubocop/ast/traversal.rb

```diff
@@ -47,7 +47,7 @@ def on_#{type}(node)
       MANY_CHILD_NODES.each do |type|
         module_eval(<<-RUBY, __FILE__, __LINE__ + 1)
           def on_#{type}(node)
-            node.children.each { |child| send(:"on_\#{child.type}", child) }
+            node.children.compact.each { |child| send(:"on_\#{child.type}", child) }
```

Comment:
`compact` isn’t used anywhere else in this file. But a few methods have `if child` instead – e.g. `#on_case` which is now (functionally) exactly identical to `#on_irange` and the other `MANY_CHILD_NODES` methods.

I guess what I’m suggesting is: Couldn’t we add `case`, `rescue`, `resbody`, `ensure`, `for` and `when` to `MANY_CHILD_NODES`?

Reply:
`const` could be added to `ONE_CHILD_NODE`, and `on_if` looks identical to `on_class`. I think this file needs a refactoring…

Reply:
@bquorning Great point. I changed the approach to alias `#on_case` instead.

Seems like you have a pretty good idea for how to make this code better. Will leave the additional improvements to you. 🙏

Reply:
I never noticed this piece of code before yesterday 😆 But when I did, I started studying and rewriting it.